include/oneapi/mkl/sparse_blas/detail/onemkl_sparse_blas_backends.hxx

```diff
@@ -20,72 +20,198 @@
-    index_base index, sycl::buffer<intType, 1> &row_ptr, sycl::buffer<intType, 1> &col_ind,
-    sycl::buffer<fpType, 1> &val);
+bool set_matrix_property(sycl::queue &queue, matrix_handle_t smhandle, matrix_property property);
 
-template <typename fpType, typename intType>
-ONEMKL_EXPORT std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>, sycl::event>
-set_csr_data(sycl::queue &queue, matrix_handle_t handle, intType num_rows, intType num_cols,
-             intType nnz, index_base index, intType *row_ptr, intType *col_ind, fpType *val,
-             const std::vector<sycl::event> &dependencies = {});
+// SPMM
+ONEMKL_EXPORT void init_spmm_descr(sycl::queue &queue, spmm_descr_t *p_spmm_descr);
 
-ONEMKL_EXPORT sycl::event optimize_gemm(sycl::queue &queue, transpose transpose_A,
-                                        matrix_handle_t handle,
-                                        const std::vector<sycl::event> &dependencies = {});
+ONEMKL_EXPORT sycl::event release_spmm_descr(sycl::queue &queue, spmm_descr_t spmm_descr,
+                                             const std::vector<sycl::event> &dependencies = {});
+
+ONEMKL_EXPORT void spmm_buffer_size(sycl::queue &queue, oneapi::mkl::transpose opA,
+                                    oneapi::mkl::transpose opB, const void *alpha,
+                                    matrix_view A_view, matrix_handle_t A_handle,
+                                    dense_matrix_handle_t B_handle, const void *beta,
+                                    dense_matrix_handle_t C_handle, spmm_alg alg,
+                                    spmm_descr_t spmm_descr, std::size_t &temp_buffer_size);
 
-ONEMKL_EXPORT sycl::event optimize_gemm(sycl::queue &queue, transpose transpose_A,
-                                        transpose transpose_B, layout dense_matrix_layout,
-                                        const std::int64_t columns, matrix_handle_t handle,
+ONEMKL_EXPORT void spmm_optimize(sycl::queue &queue, oneapi::mkl::transpose opA,
+                                 oneapi::mkl::transpose opB, const void *alpha, matrix_view A_view,
+                                 matrix_handle_t A_handle, dense_matrix_handle_t B_handle,
+                                 const void *beta, dense_matrix_handle_t C_handle, spmm_alg alg,
+                                 spmm_descr_t spmm_descr, sycl::buffer<std::uint8_t, 1> workspace);
+
+ONEMKL_EXPORT sycl::event spmm_optimize(sycl::queue &queue, oneapi::mkl::transpose opA,
+                                        oneapi::mkl::transpose opB, const void *alpha,
+                                        matrix_view A_view, matrix_handle_t A_handle,
+                                        dense_matrix_handle_t B_handle, const void *beta,
+                                        dense_matrix_handle_t C_handle, spmm_alg alg,
+                                        spmm_descr_t spmm_descr, void *workspace,
```

Comment:
So we have a `struct spmm_descr`, then `using spmm_descr_t = spmm_descr *;`, and here we have `spmm_descr_t spmm_descr`. Although this is fine, maybe we could have chosen a better name for the argument when writing the spec. I presume we won't run into code issues because of this. I'm not sure if we would have run into problems if we had `typedef struct spmm_descr` instead of just `struct spmm_descr`.